ui-components/src/page_components/listing/listing_sidebar/Waitlist.tsx

```diff
@@ -1,77 +1,46 @@
-  return props.value ? (
-    <li className={`uppercase text-gray-800 text-tiny ${props.className}`}>
-      <span className="text-right w-12 inline-block pr-2">{props.value}</span>
-      <span>{props.text}</span>
-    </li>
-  ) : null
-}
-
 export interface WaitlistProps {
   isWaitlistOpen: boolean
-  waitlistMaxSize: number
-  waitlistCurrentSize: number
-  waitlistOpenSpots: number
-  unitsAvailable: number
+  waitlistMaxSize?: number | null
+  waitlistCurrentSize?: number | null
+  waitlistOpenSpots?: number | null
 }
 
+const WaitlistItem = (props: { className?: string; value: number; text: string }) => (
+  <li className={`uppercase text-gray-800 text-tiny ${props.className}`}>
+    <span className="text-right w-12 inline-block pr-2">{props.value}</span>
+    <span>{props.text}</span>
+  </li>
+)
+
 const Waitlist = (props: WaitlistProps) => {
-  let header, subheader, waitlistItems
-  if (!props.isWaitlistOpen && !props.waitlistCurrentSize) return <></>
-  if (props.unitsAvailable && props.unitsAvailable > 0 && props.isWaitlistOpen) {
-    header = t("listings.waitlist.unitsAndWaitlist")
-    subheader = t("listings.waitlist.submitAnApplication")
-    waitlistItems = (
-      <>
-        <WaitlistItem
-          value={props.unitsAvailable}
-          text={t("listings.availableUnits")}
-          className={"font-semibold"}
-        />
-        {props.waitlistOpenSpots && (
-          <WaitlistItem
-            value={props.waitlistOpenSpots}
-            text={t("listings.waitlist.openSlots")}
-            className={"font-semibold"}
-          />
-        )}
-      </>
-    )
-  } else {
-    if (props.isWaitlistOpen) {
-      header = t("listings.waitlist.isOpen")
-      subheader = t("listings.waitlist.submitForWaitlist")
-    } else {
-      header = t("listings.waitlist.closed")
-      subheader = null
-    }
-    waitlistItems = (
-      <>
-        <WaitlistItem
-          value={props.waitlistCurrentSize || 0}
-          text={t("listings.waitlist.currentSize")}
-        />
-        {props.waitlistOpenSpots && (
-          <WaitlistItem
-            value={props.waitlistOpenSpots}
-            text={t("listings.waitlist.openSlots")}
-            className={"font-semibold"}
-          />
-        )}
-        <WaitlistItem value={props.waitlistMaxSize || 0} text={t("listings.waitlist.finalSize")} />
-      </>
-    )
-  }
+  if (!props.isWaitlistOpen) return <></>
 
   return (
     <section className="aside-block is-tinted">
-      <h4 className="text-caps-tiny">{header}</h4>
+      <h4 className="text-caps-tiny">{t("listings.waitlist.unitsAndWaitlist")}</h4>
       <div>
-        {subheader && <p className="text-tiny text-gray-800 pb-3">{subheader}</p>}
-        {<ul>{waitlistItems}</ul>}
+        <p className="text-tiny text-gray-800 pb-3">{t("listings.waitlist.submitAnApplication")}</p>
+        <ul>
+          {props.waitlistCurrentSize != null && (
```

Comment:
!==

Reply:
Ah, so this is what got me off on the wrong tangent before. 😅 When checking for something that's either null or undefined, you can't use the double-equals. Some more context here: https://stackoverflow.com/a/15992131

Reply:
In cases like this, I'd rather us explicitly check for both, so `props.waitlistCurrentSize !== null && props.waitlistCurrentSize !== undefined`

Reply:
Something that we could do (not here and for a larger point of discussion), and I've done in past projects, is to avoid using null altogether. Right now we have this mix of values being null or undefined so we always have to check for both, but we could make a system wide decision to clean up all the nulls and use undefined instead. There's likely a lot of work to do there, but it would be nice to have some consistency.